lib/PatternMultiTopicsConsumerImpl.cc

```diff
@@ -193,10 +195,10 @@ void PatternMultiTopicsConsumerImpl::onTopicsRemoved(NamespaceTopicsPtr removedT
 NamespaceTopicsPtr PatternMultiTopicsConsumerImpl::topicsPatternFilter(
     const std::vector<std::string>& topics, const PULSAR_REGEX_NAMESPACE::regex& pattern) {
     NamespaceTopicsPtr topicsResultPtr = std::make_shared<std::vector<std::string>>();
-
-    for (std::vector<std::string>::const_iterator itr = topics.begin(); itr != topics.end(); itr++) {
-        if (PULSAR_REGEX_NAMESPACE::regex_match(*itr, pattern)) {
-            topicsResultPtr->push_back(*itr);
+    for (const auto& it : topics) {
```

Comment:
In C++, we usually use `it` or `iter` to represent the iterator. However, in a range for loop, the `it` here is not an iterator. It's a reference.

```c++
// 1. Iterate
for (auto it = v.cbegin(); it != v.cend(); it++) {
    cout << *it << endl;  // dereference the iterator to a reference
}
// 2. Range-for loop
for (const auto& value : v) {
    cout << value << endl;
}
```

So it's better to name it with `topic` or `topicStr` (to avoid conflict).